client/src/components/EditProject.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import { Button, Modal, ModalHeader, ModalBody, Label, Input, ModalFooter } from 'reactstrap';
+
+class EditProject extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            modal: false,
+            backdrop: true
+        };
+        this.toggle = this.toggle.bind(this);
+    }
+
+    toggle() {
+        this.setState(prevState => ({ modal: !prevState.modal }));
+    }
+
+    handleChange = event => {
+        const updateProject = { ...this.props.project, [event.currentTarget.name]: event.currentTarget.value };
+        this.props.updateYourProject(this.props.index, this.props.project.id, updateProject);
```

Comment:
A quick note about these method names that include `Your`. This is probably too constrained given a system with multiple users/roles. I am not recommending changing it but consider in the future naming methods very carefully and consider that the user in this case is the developer using the api. The api should expose a way to delete and then the implementation would check to make sure that the role had the ability to delete other's projects (if admin) or your own project (if users match).